src/components/Input.js

```diff
@@ -20,26 +20,31 @@ export default class Settings extends Component {
 
     this.state = {
     };
+    var that = this;
   }
 
   postMessage(text) {
     // Clear the text input field
-    this._textInput.setNativeProps({ text: '' });
-    // Post the message to the database
-    fetch('http://127.0.0.1:8000/messages', {
-      method: 'POST',
-      headers: {
-        Accept: 'application/json',
-        'Content-Type': 'application/json'
-      },
-      body: JSON.stringify({
-        text: text,
-        latitude: this.props.location.latitude,
-        longitude: this.props.location.longitude,
-        userAuth: this.props.userAuth
+    if(this.props.userName) {
+      this._textInput.setNativeProps({ text: '' });
+      // Post the message to the database
+      fetch('http://127.0.0.1:8000/messages', {
+        method: 'POST',
+        headers: {
+          Accept: 'application/json',
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({
+          text: text,
+          latitude: this.props.location.latitude,
+          longitude: this.props.location.longitude,
+          userAuth: this.props.userAuth
+        })
       })
-    })
-    .then(() => { this.props.getMessages(); });
+      .then(() => { this.props.getMessages(); });
+    } else {
+      that.props.login();
```

Comment:
Instead of saving `this` in a variable, bind this to the postMessage function using `this.postMessage = this.postMessage.bind(this)` in the constructor.